From 94fa398b8ea3fa89505510e6a8f9bd87d045e724 Mon Sep 17 00:00:00 2001
From: Khalid Faisal Ansari <khalid.ansari@oss.qualcomm.com>
Date: Tue, 2 Dec 2025 10:48:13 +0530
Subject: [PATCH 1/2] FROMLIST: arm64: dts: qcom: qcs9075-iq-9075-evk: Enable
 TPM

Enable ST33HTPM TPM over SPI16 on the LeMans IoT EVK by adding the
required SPI and TPM nodes.

Signed-off-by: Khalid Faisal Ansari <khalid.ansari@oss.qualcomm.com>
Upstream-Status: Submitted [https://lore.kernel.org/all/20251114-enable-tpm-lemans-v1-1-c8d8b580ace7@oss.qualcomm.com/]
---
 arch/arm64/boot/dts/qcom/qcs9075-iq-9075-evk.dts | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/qcs9075-iq-9075-evk.dts b/arch/arm64/boot/dts/qcom/qcs9075-iq-9075-evk.dts
index 7774cc82186a..169028ac77c7 100644
--- a/arch/arm64/boot/dts/qcom/qcs9075-iq-9075-evk.dts
+++ b/arch/arm64/boot/dts/qcom/qcs9075-iq-9075-evk.dts
@@ -995,8 +995,13 @@ &spi15 {
 };
 
 &spi16 {
-	qcom,load-firmware;
-	qcom,xfer-mode = <1>;
+	status = "okay";
+
+	tpm@0 {
+		compatible = "st,st33htpm-spi", "tcg,tpm_tis-spi";
+		reg = <0>;
+		spi-max-frequency = <20000000>;
+	};
 };
 
 &spi17 {
-- 
2.34.1

